lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -700,7 +700,13 @@ def get_device_facts(self):
 
             sg_inq = self.module.get_bin_path('sg_inq')
 
-            if sg_inq:
+            # we can get NVMe device's serial number from /sys/block/<name>/device/serial
+            serial_path = "/sys/block/%s/device/serial" % (block)
```

Comment:
Are you able to read this path on your system without root?

Reply:
Yes, sure. That file is readable for everyone:
```
[ec2-user@ip-10-xx-xx-xx ~]$ id
uid=1000(ec2-user) gid=1000(ec2-user) groups=1000(ec2-user),4(adm),10(wheel),190(systemd-journal) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023

[ec2-user@ip-10-xx-xx-xx ~]$ ls -la /sys/block/nvme0n1/device/serial
-r--r--r--. 1 root root 4096 Jun 25 03:07 /sys/block/nvme0n1/device/serial

[ec2-user@ip-10-xx-x-xx ~]$ cat /sys/block/nvme0n1/device/serial
vol0901e25e812a4f8bb
```